lib/exgboost.ex

```diff
@@ -11,63 +11,33 @@ defmodule EXGBoost do
   runs on major distributed environment (Hadoop, SGE, MPI) and can solve problems beyond
   billions of examples.
 
-  ## Installation
-
-  ```elixir
-  def deps do
-  [
-    {:exgboost, "~> 0.2.1"}
-  ]
-  end
-  ```
-
   ## API Data Structures
 
-  EXGBoost's top-level `EXGBoost` API works directly and only with `Nx` tensors. However, under the hood,
-  it leverages the structs defined in the `EXGBoost.Booster` and `EXGBoost.DMatrix` modules. These structs
-  are wrappers around the structs defined in the XGBoost library.
-  The two main structs used are [DMatrix](https://xgboost.readthedocs.io/en/latest/c.html#dmatrix)
-  to represent the data matrix that will be used
-  to train the model, and [Booster](https://xgboost.readthedocs.io/en/latest/c.html#booster)
-  which represents the model.
-
-  The top-level `EXGBoost` API does not expose the structs directly. Instead, the
-  structs are exposed through the `EXGBoost.Booster` and `EXGBoost.DMatrix` modules. Power users
-  might wish to use these modules directly. For example, if you wish to use the `Booster` struct
-  directly then you can use the `EXGBoost.Booster.booster/2` function to create a `Booster` struct
-  from a `DMatrix` and a keyword list of options. See the `EXGBoost.Booster` and `EXGBoost.DMatrix`
-  modules source for more implementation details.
+  EXGBoost's top-level `EXGBoost` API works directly and only with `Nx.Tensor` for data
+  representation and with `EXGBoost.Booster` structs as an internal representation.
+  Direct manipulation of `EXGBoost.Booster` structs is discouraged.
```

Comment:
I guess this is a great question as to what level of exposure those 2 should have. The Python API actually uses DMatrix and Booster classes directly, but after discussion with Sean we decided that (at least initially) we should only have the top level API documented to encourage its use which has tensors as args rather than using DMatrix or Booster. I think the docs referring to them is from prior to us changing that philosophy.  What're your thoughts?

Reply:
Ah, this makes sense! I think those modules shouldn't be exposed either. Booster needs to be at least a little bit documented because it's how the model is represented, but you can document the module but not any functions there, for instance.

Reply:
Ok, so include a moduledoc explaining the module and perhaps some of the key functions but otherwise do doc false for the others?

Reply:
My main issue with adding docs to functions in the Booster module is that they all receive DMatrix, and so should be hidden